components/datagrid/DataGrid.js

```diff
@@ -123,6 +119,8 @@ class DataGrid extends PureComponent {
       [theme['has-border-right']]: selectable || stickyFromLeft > 0,
     });
 
+    const isPartiallySelected = selectedRows.length > 0 && selectedRows.length !== children[1].length;
```

Comment:
What's `children[1]`? Something tells me there's something fishy about this (although the code was actually already in here).

Reply:
It's indeed a bit weird. It's because the children consists of 2 or more nested arrays. The first one being the HeaderRow elements, the 2nd one being the BodyRow elements and the 3rd the FooterRow elements. In the render method where they map over the children, they check the type with isComponentOfType to render the correct element. Should I flat it out and then filter to take the length of that array?

Reply:
what about

```js

const [headerRow, bodyRow] = children;

```

for readability?

Reply:
I actually think you're a bit mistaken.
Datagrid can contain multiple `BodyRow`'s as children, for example:


```js
<DataGrid>
	<HeaderRow />

	<BodyRow />
	<BodyRow />
	<BodyRow />

	<FooterRow />
	<FooterRow />
</DataGrid>
```

So (if I'm correct), `children[1]` is actually just a single `BodyRow`. It's also not (technically) required to have a `HeaderRow` I think.

I think the best solution here is to filter the children with `isComponentOfType`.

Reply:
Hm, okay, I will make sure to filter the children, so we are prepared for the edge cases.

 :ok: by @Kemosabert on 25/07/2018 09:20:13